apps/functions-subscription/src/adapters/azure/functions/__tests__/get-subscription.test.ts

```diff
@@ -0,0 +1,58 @@
+import { describe, expect, it } from 'vitest';
+import * as TE from 'fp-ts/TaskEither';
+import { makeAValidGetSubscriptionRequest } from './data';
+import { makeFunctionContext, makeTestSystemEnv } from './mocks';
+import { aSubscription } from '../../../../domain/__tests__/data';
+import { ItemNotFound } from '../../../../domain/errors';
+import { makeGetSubscriptionHandler } from '../get-subscription';
+
+describe('get subscription', () => {
+  it('should return 404 when the subscription does not exist', async () => {
+    const env = makeTestSystemEnv();
+
+    const error = new ItemNotFound('Subscription not found');
+    env.getSubscription.mockReturnValueOnce(TE.left(error));
+
+    const actual = await makeGetSubscriptionHandler(env)(
+      makeAValidGetSubscriptionRequest(),
+      makeFunctionContext(),
+    );
+
+    expect(actual.status).toStrictEqual(404);
+    expect(await actual.json()).toMatchObject({
+      status: 404,
+      detail: error.message,
+    });
```

Comment:
**note**: The `toMatchObject` needs to verify at least some properties of the returned object.

The source code before the introduction of the `toHttpProblemJson` function led to the tests pass even if the payload of the response wasn't what we expected.
We checked only the HTTP status code, but in case of errors, the status code of the request and the property `status` of the response body were different.

The `toMatchObject` led the tests fail in case of these discrepancies.